pkg/pgmodel/lreader/labels_reader.go

```diff
@@ -78,9 +80,79 @@ func (lr *labelsReader) LabelValues(labelName string) ([]string, error) {
 	return labelValues, nil
 }
 
+const (
+	getLabelIDofValidTenant           = `SELECT id::INTEGER FROM _prom_catalog.label WHERE key = '__tenant__' AND value = $1::TEXT`
+	labelCompareClause                = ` $%d::INTEGER = any(labels) `
+	getLabelNamesBelongingToTenantIDs = `
+SELECT ARRAY_AGG(a.key) FROM
```

Comment:
Can we use the following query which hopefully gives the expected result in a single query.?

```
select distinct key from "_prom_catalog".label where id in (select unnest(labels) as valid_label_ids from "_prom_catalog".series where labels && (SELECT array_agg(id::INTEGER) FROM _prom_catalog.label WHERE key = '__tenant__' and value=any(array['foo', 'bar'])));
```
'foo' and 'bar' are the tenants here.